clouddrift/binning.py

```diff
@@ -0,0 +1,177 @@
+"""Module for binning Lagrangian data."""
+
+from collections.abc import Iterable
+
+import numpy as np
+import xarray as xr
+
+DEFAULT_BINS_NUMBER = 10
+
+
+def binned_statistics(
+    coords: np.ndarray | list[np.ndarray],
+    data: np.ndarray | list[np.ndarray] | None = None,
+    bins: int | list = DEFAULT_BINS_NUMBER,
+    bins_range: list | None = None,
+    dim_names: list[str] | None = None,
+    output_names: list[str] | None = None,
+    zeros_to_nan: bool = False,
+) -> xr.Dataset:
+    """
+    Compute N-dimensional histogram binning and calculate variable means in each bin.
+
+    Parameters
+    ----------
+    coords : array-like or list of array-like
+        Arrays of the coordinates of the Lagrangian data to be binned. For 1D data, a single array can be provided.
+        For multiple dimensions, the first array represents each coordinates along the first dimension,
+        the second array represents each coordinates along the second dimension, and so on.
+    data : array-like or list of array-like
+        Data associated at the Lagrangian coordinates described in coords. Multiple variables can be provided as a list.
```

Comment:
Data values associated with the Lagrangian coordinates in coords. Can be a single array or a list of arrays for multiple variables